packages/compiler/src/typekit/kits/builtin.ts

```diff
@@ -142,6 +148,36 @@ declare module "../define-kit.js" {
 
 defineKit<TypekitExtension>({
   builtin: {
+    is(type: Type): boolean {
+      // Model property does not have a namespace,
+      // so we will use the Model where it is defined.
+      if (type.kind === "ModelProperty" && type.model) {
```

Comment:
not sure this is worth it?

Reply:
Updated the tests to show when this is helpful 

```tsp
namespace TypeSpec {
        model InTypeSpec {
          foo: string;
        }
      }

      model Foo {
        bar: TypeSpec.InTypeSpec.foo;
      } 
```

Without this $.builtin.is(Foo.bar.type) would return true